File: Core/Entities/Hardware.cs
namespace Core.Entities
{
    public class Hardware : BaseEntity
    {
        public string Name { get; set; }
        public HardwareType HardwareType { get; set; }
        public int HardwareTypeId { get; set; }
        public int Size { get; set; }
        public HardwareMaterial HardwareMaterial { get; set; }
        public int HardwareMaterialId { get; set; }
        public HardwareColor HardwareColor { get; set; }
        public int HardwareColorId { get; set; }
        public int InStock { get; set; }
        public int Ordered { get; set; }
        public decimal Price { get; set; }
    }
}

File: client/src/app/models/hardware.ts
export interface IHardware {
    id: number;
    name: string;
    hardwareType: string;
    hardwareTypeId: number;
    size: number;
    hardwareMaterial: string;
    hardwareMaterialId: number;
    hardwareColor: string;
    hardwareColorId: number;
    inStock: number;
    ordered: number;
    price: number;
}

export interface IHardwareToCreate {
    name: string;
    hardwareTypeId: number;
    size: number;
    hardwareMaterialId: number;
    hardwareColorId: number;
    inStock: number;
    ordered: number;
    price: number;
}

export class HardwareFormValues implements IHardwareToCreate {
    name: string;
    hardwareTypeId: number;
    size = 0;
    hardwareMaterialId: number;
    hardwareColorId: number;
    inStock = 0;
    ordered = 0;
    price = 0;

    constructor(init?: HardwareFormValues) {
        Object.assign(this, init);
    }
}